klippy/extras/palette2.py

```diff
@@ -215,6 +218,20 @@ def cmd_OmegaDefault(self, gcmd):
         if self._check_P2(gcmd):
             self.write_queue.put(gcmd.get_commandline())
 
+    def _wait_for_heartbeat(self):
+        startTs = self.reactor.monotonic()
+        while self.heartbeat is None and self.heartbeat < (
+                self.reactor.monotonic() -
+                SETUP_TIMEOUT) and startTs > (
+                self.reactor.monotonic() -
+                SETUP_TIMEOUT):
+            self.reactor.pause(1)
```

Comment:
I know it's not related to this patch, but in review I noticed at error.  The `pause()` method takes an absolute time not a duration.  So, it should be something like `self.reactor.pause(curtime + 1.)`.

Also, as a minor note, for efficiency reasons it's possible to obtain a new time directly from `pause()` - as in `curtime = self.reactor.pause(curtime + 1.)`